modules/compiler/utils/source/replace_local_module_scope_variables_pass.cpp

```diff
@@ -236,12 +236,9 @@ PreservedAnalyses compiler::utils::ReplaceLocalModuleScopeVariablesPass::run(
     // then add our element type to the struct
     structElementTypes.push_back(memberType);
 
-    // update the offset, taking into account any array length
-    unsigned int totalSize = alignment;
-    while (memberType->isArrayTy()) {
-      totalSize *= memberType->getArrayNumElements();
-      memberType = memberType->getArrayElementType();
-    }
+    // update the offset based on the type's size
+    unsigned int totalSize =
+        M.getDataLayout().getTypeAllocSize(memberType).getFixedValue();
```

Comment:
This should possibly be [`getTypeStoreSize`](https://llvm.org/doxygen/classllvm_1_1DataLayout.html#acfcd22eb38dbfe1acbf138754297437a) instead? `getTypeAllocSize` includes the padding that would be required if we were to add another instance of this type right after.

Reply:
I think getTypeAllocSize is correct here.

Doing some testing, a struct `{i20, i20}` is 8 bytes wide, so an individual `i20` entry in that struct must be 4 bytes, which includes the 12 bits of padding. That is the value that getTypeAllocSize returns (getTypeStoreSize only returns 3 bytes).